src/clients/HubPoolClient.ts

```diff
@@ -452,21 +454,42 @@ export class HubPoolClient {
         );
       }
     ) as ExecutedRootBundle;
-    if (mostRecentExecutedRootBundleEvent) {
-      // Arguably we don't need to even check these array lengths since we should assume that any proposed root bundle
-      // meets this condition.
-      if (
-        mostRecentExecutedRootBundleEvent.l1Tokens.length !== mostRecentExecutedRootBundleEvent.runningBalances.length
-      ) {
-        throw new Error("runningBalances and L1 token of ExecutedRootBundle event are not same length");
+
+    if (executedRootBundle) {
+      // runningBalances length must be a 1x or 2x multiple of l1Tokens length.
+      const { l1Tokens } = executedRootBundle;
+      let { runningBalances } = executedRootBundle;
+      let incentiveBalances: BigNumber[] = [];
+      if (runningBalances.length === l1Tokens.length) {
+        // Pre-UBA model (no incentives exist).
```

Comment:
I think this check should be a bit different:
- If `ConfigStore.Version < 2` (for example), then `assert(runningBalances.length == l1Tokens.length)`
- Else `assert(runningBalances.length == l1Tokens.length * 2)`

Reply:
This makes sense; have implemented it here: https://github.com/across-protocol/sdk-v2/pull/213/commits/c0bafa5372d181fcbab8b12d1143ff429b856231

It now handles this even earlier in the process - i.e. when each ExecutedRootBundle event is parsed in the update() method. It does as such involve extending the ExecutedRootBundle event to have an incentiveBalances array; this should be backwards compatible with existing uses of ExecutedRootBundle though.